File: src/test/java/kitchenpos/eatinorders/tobe/OrderTableTest.java
package kitchenpos.eatinorders.tobe;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;
import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class OrderTableTest {

    @DisplayName("주문 테이블 생성.")
    @Test
    void create() {
        OrderTableName name = new OrderTableName("테이블 명");
        assertDoesNotThrow(() -> new OrderTable(name));
    }

    @DisplayName("주문 테이블을 생성한다. 초기 생성시 손님 숫자는 0명, 공석으로 생성된다.")
    @Test
    void init_value() {
        OrderTable orderTable = orderTable();

        assertAll(
                () -> assertThat(orderTable.isEmpty()).isTrue(),
                () -> assertThat(orderTable.getNumberOfGuests().getNumberOfGuests()).isZero()
        );
    }

    @DisplayName("주문 테이블의 empty 값이 true 일때 착석 요청시 예외.")
    @Test
    void sit_exception() {
        OrderTable sut = orderTable();
        sut.sit();

        assertThatThrownBy(() -> sut.sit()).isInstanceOf(IllegalArgumentException.class);
    }

    @DisplayName("주문 테이블에 착성 요청시 empty 값이 false 로 변경 된다.")
    @Test
    void sit() {
        OrderTable sut = orderTable();
        sut.sit();

        assertThat(sut.isEmpty()).isFalse();
    }

    private OrderTable orderTable() {
        OrderTableName name = new OrderTableName("테이블 명");
        return new OrderTable(name);
    }
}

File: src/test/java/kitchenpos/eatinorders/tobe/OrderTableTest.kt
package kitchenpos.eatinorders.tobe

import kitchenpos.eatinorders.tobe.domain.OrderTableGuestNumber
import kitchenpos.fixture.EatInOrderFixtures.orderTable
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class OrderTableTest {
    @Test
    fun `주문테이블 생성`() {
        val orderTable = orderTable()

        assertThat(orderTable.numberOfGuests).isEqualTo(OrderTableGuestNumber(0))
        assertThat(orderTable.occupied).isFalse()
    }

    @Test
    fun `주문 테이블의 좌석 수 변경`() {
        val orderTable = orderTable()

        orderTable.sit()
        orderTable.changeNumberOfGuest(OrderTableGuestNumber(10))

        assertThat(orderTable.numberOfGuests).isEqualTo(OrderTableGuestNumber(10))
    }

    @Test
    fun `미점유 중인 주문 테이블의 좌석 수 변경 불가능`() {
        val orderTable = orderTable()

        assertThatThrownBy {
            orderTable.changeNumberOfGuest(OrderTableGuestNumber(10))
        }.isInstanceOf(IllegalArgumentException::class.java).withFailMessage("미점유 중인 테이블의 고객 좌석 수를 변경할 수 없습니다")
    }
}
